content/en/details/skbuild.md

```diff
@@ -15,13 +15,129 @@ DPC++ is an LLVM-based compiler, but the host portion of the binary it produces
+on writing data-parallel code in DPC++ while automating generation of necessary Python bindings.
+
+#### Building packages with setuptools
+
+When using setuptools we [used][kde-setuptools] environment variables `CC` and `LDSHARED` recognized by `setuptools` to ensure that `dpcpp` is used to compile and link extensions. 
+
+The resulting extension is a fat binary, containing both the host code with Python bindings and offloading orchestration, and the device code usually stored 
+in cross-platform intermediate representation (SPIR-V) and compiled for the device indicated via the execution queue argument using tooling from compute runtime.
+
+#### Building packages with scikit-build
+
+Using setuptools is convenient, but feels klunky. Enter [scikit-build][scikit-build].
+
+Scikit-build enables writing the logic of Python package building in CMake which [supports oneAPI DPC++][cmake-dpcpp]. Scikit-build supports building of both
+Cython-generated and pybind11-generated native extensions. `dpctl` integration with CMake allows to conveniently using `dpctl` integration with these extension generators
+simply by including 
+
+```cmake
+find_package(Dpctl REQUIRED)
+```
+
+In order for CMake to locate the script that would make this work the example `CMakeLists.txt` in [`kde_skbuild`][kde-skbuild] package implements `DPCTL_MODULE_PATH` variable 
+which can be set to output of `python -m dpctl --cmakedir`. Integration of DPC++ with CMake requires that CMake's C and/or C++ compiler were set to Intel LLVM compilers
+provided in oneAPI base kit.
 
 ```bash
-CXX=dpcpp LDSHARED="dpcpp --shared" python setup.py develop
+python setup.py develop -G Ninja -- \
+    -DCMAKE_C_COMPILER=icx          \
+    -DCMAKE_CXX_COMPILER=icpx       \
+    -DDPCTL_MODULE_PATH=$(python -m dpctl --cmakedir)
+```
+
+Altenatively, we can rely on CMake recognizing `CC` and `CXX` environment variables to shorten the input
+
+```
+CC=icx CXX=icpx python setup.py develop -G Ninja -- -DDCPTL_MODULE_PATH=$(python -m dpctl --cmakedir)
+```
+
+
+Whichever way of build the data-parallel extension apeals to you, the end result allows to offload certain computations for any of the supported devices:
+
+```ipython
+import dpctl
+import numpy as np
+import kde_skbuild as kde 
+
+cpu_q = dpctl.SyclQueue("cpu")
+gpu_q = dpctl.SyclQueue("gpu")
+
+# output info about targeted devices
+cpu_q.print_device_info()
+gpu_q.print_device_info()
+
+x = np.linspace(0.1, 0.9, num=14000)
+data = np.random.uniform(0, 1, size=10**6)
+
+# Notice that first evaluation results in JIT-compiling the kernel
+# Subsequent evaluation reuse cached binary
+f0 = kde.cython_kde_eval(cpu_q, x[:, np.newaxis], data[:, np.newaxis], 3e-6)
+
+f1 = kde.cython_kde_eval(gpu_q, x[:, np.newaxis], data[:, np.newaxis], 3e-6)
+
+assert np.allclose(f0, f1)
 ```
 
+Of course a naive NumPy reference implementation can not handle the large size input like used above, running out of memory,
```

Comment:
reads a bit awkward. do you mean Numpy will run out of memory?

Reply:
Yes with 

```
MemoryError: Unable to allocate 104. GiB for an array with shape (14000, 1000000, 1) and data type float64
```